rules/xcode_sdk_frameworks.bzl

```diff
@@ -0,0 +1,446 @@
+
+    return {
+        "ios": _search_sdk_version_from_output(xcodebuild_result.stdout, "iphoneos"),
+        "macos": _search_sdk_version_from_output(xcodebuild_result.stdout, "macosx"),
+    }
+
+def _create_xcode_framework_targets(
+        repository_ctx,
+        xcode_version_name,
+        developer_dir,
+        xcode_version):
+    """Creates a BUILD file for all the SDK frameworks contained in a given Xcode version."""
+
+    developer_dir_path = repository_ctx.path(developer_dir)
+    platforms_dir = developer_dir_path.get_child("Platforms")
+    versions = _sdk_versions_by_platforms(repository_ctx, developer_dir)
+    xcode_version_folder = repository_ctx.path(xcode_version_name)
+    host_cpu = repository_ctx.os.arch
+    for platform_name, target_name in PLATFORM_TUPLES:
+        platform_dir = platforms_dir.get_child(platform_name + ".platform")
+        platform_version = versions[target_name]
+
+        sdk_path = platform_dir.get_child("Developer").get_child("SDKs").get_child("{}.sdk".format(platform_name))
+        target_triple = _platform_target_triple(host_cpu, platform_name, target_name, platform_version)
+
+        output_folder = xcode_version_folder.get_child(platform_name)
+        overrides = _sdk_overrides(platform_name, xcode_version)
+        _create_build_file_for_sdk(
+            repository_ctx = repository_ctx,
+            developer_dir = developer_dir_path,
+            sdk_path = sdk_path,
+            output_folder = output_folder,
+            target_triple = target_triple,
+            overrides = overrides,
+        )
+
+    xcode_version_build_file_path = xcode_version_folder.get_child("BUILD.bazel")
+    repository_ctx.file(
+        xcode_version_build_file_path,
+        XCODE_VERSION_BUILD_FILE_TMPL.format(xcode_version = xcode_version_name),
+    )
+
+def _stub_frameworks(repository_ctx):
+    repository_ctx.file(
+        "BUILD.bazel",
+        content = """package(default_visibility = ["//visibility:public"])
+
+objc_library(
+    name = "xcode_sdk_frameworks",
+)
+
+""",
+    )
+
+def _xcode_sdk_frameworks_impl(repository_ctx):
+    os_name = repository_ctx.os.name.lower()
+
+    # Only supports on MacOS
+    if not os_name.startswith("mac os x"):
+        return
+
+    use_explicit_modules = repository_ctx.os.environ.get("EXPLICIT_MODULES", False)
+
+    if not use_explicit_modules:
+        _stub_frameworks(repository_ctx)
+        return
+
+    # Locates all local xcodes
+    xcode_toolchains, error = run_xcode_locator(
+        repository_ctx,
+        Label(repository_ctx.attr.xcode_locator),
+    )
+    if error:
+        fail("Failed to locate xcode with error: {}".format(error))
+
+    config_setting_lines = []
+    framework_select_lines = []
+
+    # Configures SDK frameworks for each xcode version.
+    for xcode_toolchain in xcode_toolchains:
+        xcode_version_name = "version{}".format(xcode_toolchain.version.replace(".", "_"))
+        _create_xcode_framework_targets(
```

Comment:
how many xcode versions do you have locally? so far i got it green with 15.4 only via:
```
        if not xcode_version_name.startswith("version15_4"):
            continue
```
with 15.1 or 15.2 i got:
```
ERROR: /private/var/tmp/_bazel_yuanfeng/8f12d9050b931ac3050ca4cf9fc59845/external/_main~xcode_sdk_frameworks~xcode_sdk_frameworks/version15_1_0_15C65/iPhoneOS/BUILD.bazel:7:19: Label '@@_main~xcode_sdk_frameworks~xcode_sdk_frameworks//version15_1_0_15C65/iPhoneOS:CoreMotion_swift' is duplicated in the 'deps' attribute of rule 'bazel_xcode_imports_swift'
ERROR: /private/var/tmp/_bazel_yuanfeng/8f12d9050b931ac3050ca4cf9fc59845/external/_main~xcode_sdk_frameworks~xcode_sdk_frameworks/version15_1_0_15C65/iPhoneOS/BUILD.bazel:7:19: Label '@@_main~xcode_sdk_frameworks~xcode_sdk_frameworks//version15_1_0_15C65/iPhoneOS:CoreAudio_swift' is duplicated in the 'deps' attribute of rule 'bazel_xcode_imports_swift'
ERROR: package contains errors: version15_1_0_15C65/MacOSX
ERROR: package contains errors: version15_1_0_15C65/iPhoneOS
```
and i cannot build with xcode 16 due to the error above....

Reply:
I didn't test it on 15.1 or 15.2. Thanks for the heads up! We can address the issue with different xcode versions later.